vitest.config.ts

```diff
@@ -0,0 +1,10 @@
+import { defineConfig } from "vitest/config";
+import path from "path";
+
+export default defineConfig({
+    test: {
+        alias: {
+            obsidian: path.resolve(__dirname, "mocks/obsidian.ts"),
```

Comment:
`obsidian` needs to be mocked. The functions I have tested in this PR does not depend on `obsidian`, so it is currently an empty file.

See: https://github.com/vitest-dev/vitest/issues/4029#issuecomment-1837183186